node/src/actors/chain_manager/mod.rs

```diff
@@ -1026,22 +1034,25 @@ fn update_pools(
     rep_info
 }
 
-fn separate_honest_liars<K, V, I>(rep_info: I) -> (Vec<K>, Vec<(K, V)>)
+fn separate_honest_errors_and_liars<K, I>(rep_info: I) -> (Vec<K>, Vec<K>, Vec<(K, u32)>)
 where
-    V: Default + PartialEq,
-    I: IntoIterator<Item = (K, V)>,
+    I: IntoIterator<Item = (K, RequestResult)>,
 {
     let mut honests = vec![];
     let mut liars = vec![];
-    for (pkh, num_lies) in rep_info {
-        if num_lies == V::default() {
+    let mut errors = vec![];
+    for (pkh, result) in rep_info {
+        if result.lies > 0 {
+            liars.push((pkh, result.lies));
+        // TODO: Decide which percentage would be fair enough
```

Comment:
The behavior implemented in this PR is to only give reputation to nodes whose number of errors is smaller than the number of "honests".

This opens an attack vector where an attacker can flood the network with requests that always return errors, and nobody will be able to earn reputation except of course the attacker nodes, which will avoid participating in their own requests and will slowly accumulate all the reputation.

Reply:
This attack need several wits to pay the rewards, because witness will earn the wits unless they do not earn reputation. The other option, give reputation to error_committers could produce incentives for node operators to commit errors without the risk to lose reputation, but also wining it

Reply:
Currently if one identity commits one error and one "honest", it will not be rewarded any reputation. Maybe we can change the `>` into `>=` to be less aggressive.

Reply:
Should we open an issue to further assess which percentage would be fair enough?